assembly/src/ast/mod.rs

```diff
@@ -602,6 +640,52 @@ impl Deserializable for ProcedureAst {
     }
 }
 
+/// Represents a re-exported procedure.
+///
+/// A re-exported procedure is a procedure that is defined in a different module in the same
+/// library or a different library and re-exported with the same or a different name. The
+/// re-exported procedure is not copied into the module, but rather a reference to it is added to
+/// the [ModuleAST].
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct ProcReExport {
+    name: ProcedureName,
+    ref_path: LibraryPath,
+}
```

Comment:
Should this be more like:
```Rust
pub struct ProcReExport {
    proc_id: ProcedureId,
    alias: ProcedureName,
}
```
Basically, when we parse a re-export, we should be able to build a procedure ID for the referenced procedure based on the data we have. Any reason to keep the full `LibraryPath` around for it?

Also, if we do it this way, I'd add something like `get_alias_Id()` method:
```Rust
impl ProcReExport {
    pub fn get_alis_id(&self, module_path: &LibraryPath) -> ProcedureId {
        ...
    }
}
```

Reply:
Yeah, `LibraryPath` is not required. It was kept initially when we didn't expect the dependency to be loaded into the assembler. One question is if instead of `alias` it should be `name` in this context instead?

Reply:
Yep, I think `name` could work even better here.